CharlieBackend.Api.UnitTes/DashboardServiceTest.cs

```diff
@@ -1,484 +1,882 @@
-
-            dashbordrepositoryMock.Setup(x => x.GetGroupsIdsByCourseIdAndPeriodAsync(
-                (long)studentResultRequestWithData.CourseId, studentResultRequestWithData.StartDate, studentResultRequestWithData.FinishDate))
-                .ReturnsAsync(new List<long>() { 2 });
-
-            dashbordrepositoryMock.Setup(x => x.GetGroupsIdsByCourseIdAndPeriodAsync(
-                (long)studentResultRequestWithOutStudent.CourseId, 
+            _dashboardRepositoryMock.Setup(x => x.GetGroupsIdsByCourseIdAndPeriodAsync(
+                (long)studentResultRequestWithOutStudent.CourseId,
                 studentResultRequestWithOutStudent.StartDate,
                 studentResultRequestWithOutStudent.FinishDate))
-                .ReturnsAsync(new  List<long>());
-
-            dashbordrepositoryMock.Setup(x => x.GetStudentsIdsByGroupIdsAsync(new List<long>() { 2 }))
-                .ReturnsAsync(new List<long>() { 6, 7, 8, 9, 10 });
-
-            dashbordrepositoryMock.Setup(x => x.GetStudentsIdsByGroupIdsAsync(new List<long>()))
                 .ReturnsAsync(new List<long>());
-            
-            dashbordrepositoryMock.Setup(x => x.GetStudentAverageMarksByStudentIdsAndGropsIdsAsync(
-               new List<long>() { 6, 7, 8, 9, 10 }, new List<long>() { 2 }))
-                .ReturnsAsync(new List<AverageStudentMarkDto>() {
-                    new AverageStudentMarkDto()
-                    {
-                        CourseId = 2,
-                        StudentGroupId = 2,
-                        StudentId = 6,
-                        StudentAverageMark = (decimal)5.1
-                    }
-                });
 
-            dashbordrepositoryMock.Setup(x => x.GetStudentAverageMarksByStudentIdsAndGropsIdsAsync(new List<long>(), new List<long>()))
+            _dashboardRepositoryMock.Setup(x => x.GetStudentsIdsByGroupIdsAsync(new List<long>()))
+               .ReturnsAsync(new List<long>());
+
+            _dashboardRepositoryMock.Setup(x => x.GetStudentAverageMarksByStudentIdsAndGropsIdsAsync(new List<long>(), new List<long>()))
                 .ReturnsAsync(new List<AverageStudentMarkDto>());
 
-            dashbordrepositoryMock.Setup(x => x.GetStudentsAverageVisitsByStudentIdsAndGroupsIdsAsync(
-                 new List<long>() { 6, 7, 8, 9, 10 }, new List<long>() { 2 }))
-                .ReturnsAsync(new List<AverageStudentVisitsDto>()
+            //Act
+            var resultWithoutStudent = await _dashboardService.GetStudentsResultAsync(studentResultRequestWithOutStudent);
+
+            //Assert
+            resultWithoutStudent.Data.AverageStudentsMarks.Should().BeNullOrEmpty();
+        }
+
+        [Fact]
+        public async Task GetStudentsResult_NotExistingCourseId_ShouldReturnEmptyStudentsAverageVisits()
+        {
+            //Arrange
+            var studentResultRequestWithOutStudent = new StudentsRequestDto<StudentResultType>()
+            {
+                CourseId = id,
```

Comment:
Since this is the test for "non existent course id" case, I think it's fine if we declare a separate local variable like
```cs
int nonExistentCourseId = int.MaxValue;
```
and pass it here in this line instead of using a common `id` field that you created. 

This way you're highlighting that it is an invalid value that is tested here, and making the purpose of the test more obvious. This way you're improving readability. 

I hope this makes sense :) Please consider doing that. And update in other tests where it is similar.